### ISSUE TITLE
about data prepare

### ISSUE THREAD

**Issue by garymmi** [MycroftAI/mycroft-precise#97]
I read the doc https://github.com/MycroftAI/mycroft-precise/wiki/Training-your-own-wake-word#how-to-train-your-own-wake-word/
It said making sure to leave a second or two of silence at the start of each recording, but with no silence after the wake word.

1. it means i can leave one second or two second of silence at beginning of a recoding?
    if yes and leave two second, when getting mfcc, only processed 1.5 sec audio file, strange.
2. if leave one second, at beginning of a recoding, it means only 0.5 sec is wake up word ?
3. how to process lots of recodings to prepare training data?
    should i do VAD and then padding silence at begin and cut silence at end?

**Comment by MatthewScholefield** [comment]
Data is processed as follows:
 - If longer than 1.5 seconds: chop to 1.5 seconds
 - If shorter than 1.5 seconds: add perfect silence in beginning to make it 1.5 seconds

Since in reality there's static in microphones, when recordings are too short and it inserts perfect silence that could degrade the real world performance of the model. Having too much audio before the wake word should never be a problem since it chops it to the correct size.

Let me know if this males sense

**Comment by MatthewScholefield** [comment]
No, the last 1.5 seconds of audio are kept and the beginning of the audio is removed.

**Comment by garymmi** [comment]
if i want to shorten 1.5 second, what codes need to be modified?
Thank you

**Comment by MatthewScholefield** [comment]
Change `buffer_t=1.5` in `precise/params.py`
